### ISSUE TITLE
0.12 thinks that removed fields are present in config

### ISSUE THREAD

**Issue by danawillow** [hashicorp/terraform#19955]
### Terraform Version

```
Terraform v0.12.0-dev
+ provider.google (unversioned)
```
Running with hashicorp/terraform at HEAD (for both the compiled binary and the vendored sdk in the google provider)

### Terraform Configuration Files

```hcl
data "google_compute_image" "my_image" {
  family  = "debian-9"
  project = "debian-cloud"
}

resource "google_compute_backend_service" "lipsum" {
  name        = "shjfgdklak"
  description = "Hello World 1234"
  port_name   = "http"
  protocol    = "HTTP"
  timeout_sec = 10

  backend {
    group = "${google_compute_instance_group_manager.foobar.instance_group}"
  }

  health_checks = ["${google_compute_http_health_check.default.self_link}"]
}

resource "google_compute_instance_group_manager" "foobar" {
  name               = "dfgjosihkl"
  instance_template  = "${google_compute_instance_template.foobar.self_link}"
  base_instance_name = "foobar"
  zone               = "us-central1-f"
  target_size        = 1
}

resource "google_compute_instance_template" "foobar" {
  name         = "rejigkhsfdnsl"
  machine_type = "n1-standard-1"

  network_interface {
    network = "default"
  }

  disk {
    source_image = "${data.google_compute_image.my_image.self_link}"
    auto_delete  = true
    boot         = true
  }
}

resource "google_compute_http_health_check" "default" {
  name               = "ejkgshdfn"
  request_path       = "/"
  check_interval_sec = 1
  timeout_sec        = 1
}
```

### Debug Output

https://gist.github.com/danawillow/0221ab4c06a129564b1a7c1a0d34a8ff

### Expected Behavior

It should have applied the config

### Actual Behavior

```
Error: "auto_healing_policies": [REMOVED] This field is in beta. Use it in the the google-beta provider instead. See https://terraform.io/docs/providers/google/provider_versions.html for more details.

  on main.tf line 20, in resource "google_compute_instance_group_manager" "foobar":
  20: resource "google_compute_instance_group_manager" "foobar" {



Error: "version": [REMOVED] This field is in beta. Use it in the the google-beta provider instead. See https://terraform.io/docs/providers/google/provider_versions.html for more details.

  on main.tf line 20, in resource "google_compute_instance_group_manager" "foobar":
  20: resource "google_compute_instance_group_manager" "foobar" {



Error: "rolling_update_policy": [REMOVED] This field is in beta. Use it in the the google-beta provider instead. See https://terraform.io/docs/providers/google/provider_versions.html for more details.

  on main.tf line 20, in resource "google_compute_instance_group_manager" "foobar":
  20: resource "google_compute_instance_group_manager" "foobar" {
```
However, none of the removed fields are present in my config.

### Steps to Reproduce

See https://github.com/hashicorp/terraform/issues/19754#issuecomment-452919515. The error in tests and also when running a compiled binary locally.

**Comment by apparentlymart** [comment]
@jbardin was this fixed by #19988, or is there still some work remaining here?

**Comment by jbardin** [comment]
Oh yes, it was. I forgot the "fixes" keyword in the pr.

**Comment by betsy-lichtenberg** [comment]
Is this bug fixed in 0.11? I seem to be hitting it when using this tutorial.

Terraform v0.11.11

https://cloud.google.com/community/tutorials/modular-load-balancing-with-terraform#internal-tcp-load-balancer-with-regional-forwarding-rule

**Comment by jbardin** [comment]
Hi @betsy-lichtenberg,

This issue is only relevant for 0.12. My only guess is an incorrect provider version, but I would also check the Google provider issue tracker to see if there are similar reports to what you're seeing.

**Comment by betsy-lichtenberg** [comment]
Thanks. I didn't see any existing tickets, so I filed this one:

https://github.com/GoogleCloudPlatform/community/issues/739

I'm new at Terraform, but digging around I found only one buried instance where the provider is defined.

/terraform-google-examples/example-lb-internal/.terraform/modules/565becfe466df58a40f46dadc26f767a/examples/simple/.terraform/modules/db73f98765a8b147eb9c43aac6ba80c6/GoogleCloudPlatform-terraform-google-managed-instance-group-215bf1d/main.tf

In my local copy, I tried changing the provider to google-beta, but that didn't fix it. It's probably more complicated than a simple fix.
